instrumentation/opentelemetry_req/lib/opentelemetry_req.ex

```diff
@@ -86,14 +207,31 @@ defmodule OpentelemetryReq do
   end
 
   defp end_span({request, %Req.Response{} = response}) do
-    attrs =
-      Map.put(%{}, Trace.http_status_code(), response.status)
-      |> maybe_append_resp_content_length(response)
+    config = Req.Request.get_private(request, :otel)
 
-    Tracer.set_attributes(attrs)
+    opt_in =
+      %{
+        HTTPAttributes.http_response_body_size() => extract_response_body_size(response)
+      }
+      |> Map.take(config.opt_in_attrs)
 
     if response.status >= 400 do
-      OpenTelemetry.Tracer.set_status(OpenTelemetry.status(:error, ""))
+      Tracer.set_status(OpenTelemetry.status(:error, ""))
```

Comment:
Looks like it was already like this, but IMO it shouldn't be an error span necessarily just because the server successfully returned a 404 for example, if that's what the caller expected. In that case, the client did its job correctly, so there was no error. In fact, the server that the client called probably did as well unless it was a 5xx error, so it shouldn't be an error on its span either. Does OTel say we should do it this way for HTTP clients, or is this just something we've chosen to do here?

Reply:
It's an error for HTTP clients.

> For HTTP status codes in the 4xx range span status MUST be left unset in case of SpanKind.SERVER
and SHOULD be set to Error in case of SpanKind.CLIENT.

https://hexdocs.pm/opentelemetry_semantic_conventions/http-spans.html#status

Reply:
IMO that's a bummer and not something I would want to have in my traces personally, so I guess that's why it's a `SHOULD` instead of a `MUST`. But as a library, I guess we have to decide whether we think it's right to follow that convention, since it's not required. My vote would be to not set it as an error unless it was like a timeout or connection error or something where there's no response at all. If it's a 5xx, I would be fine with it being marked as an error in the client, just to call attention to it even though it's not Req's fault if the server returns a 504 or something, but I think it's totally normal for 404s to happen all the time when you're just querying an API for "does this thing exist?" and it just says "no" as a 4040 - you don't need to call special attention to that.

Reply:
Users are allowed to mark a span ok so they can hook into the request and mark it should they choose to. We could look at adding a response step to let folks supply a function but this PR isn't seeking to change any existing functionality. I think that's a fine feature request.

Reply:
Yep, that makes sense! ✅